tests/test_minimal/test_utils/test_get_schema_from_method_signature.py

```diff
@@ -0,0 +1,264 @@
+from pathlib import Path
+from typing import Dict, List, Literal, Optional, Union
+
+from jsonschema import validate
+from pydantic import DirectoryPath, FilePath
+
+from neuroconv.datainterfaces import AlphaOmegaRecordingInterface
+from neuroconv.utils import get_json_schema_from_method_signature
+
+
+def test_get_json_schema_from_method_signature_basic():
+    def basic_method(
+        integer: int,
+        floating: float,
+        string_or_path: Union[Path, str],
+        boolean: bool,
+        literal: Literal["a", "b", "c"],
+        dictionary: Dict[str, str],
+        string_with_default: str = "hi",
+        optional_dictionary: Optional[Dict[str, str]] = None,
+    ):
+        pass
+
+    test_json_schema = get_json_schema_from_method_signature(method=basic_method)
+    expected_json_schema = {
+        "additionalProperties": False,
+        "properties": {
+            "boolean": {"type": "boolean"},
+            "dictionary": {"additionalProperties": {"type": "string"}, "type": "object"},
+            "floating": {"type": "number"},
+            "integer": {"type": "integer"},
+            "literal": {"enum": ["a", "b", "c"], "type": "string"},
+            "optional_dictionary": {
+                "anyOf": [{"additionalProperties": {"type": "string"}, "type": "object"}, {"type": "null"}],
+                "default": None,
+            },
+            "string_or_path": {"anyOf": [{"format": "path", "type": "string"}, {"type": "string"}]},
+            "string_with_default": {"default": "hi", "type": "string"},
+        },
+        "required": ["integer", "floating", "string_or_path", "boolean", "literal", "dictionary"],
+        "type": "object",
+    }
+
+    assert test_json_schema == expected_json_schema
+
+
+def test_get_json_schema_from_method_signature_advanced():
+    """
+    These are annotations newly supported by the Pydantic-based inference.
+
+    They should also be compatible with __future__.annotations for SpikeInterface.
+    """
+
+    # TODO: enable | usage cases when 3.11 is minimal
+    def advanced_method(
+        old_list_of_strings: List[str],
+        # new_list_of_strings: list[str],
```

Comment:
Right you are; it's only the `|` as `Union` that awaits in the future

Now that we support `list` instead of `List` I'll add a subsequent PR updating to the more modern annotation everywhere

Reply:
Yei!